ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteConnectorExecutor.java

```diff
@@ -99,6 +102,15 @@ private Tuple<Integer, Integer> calculateChunkSize(TextDocsInputDataSet textDocs
                 return Tuple.tuple(textDocsLength / stepSize + 1, stepSize);
             }
         } else {
+            Optional<ConnectorAction> predictAction = getConnector().findPredictAction();
+            if (predictAction.isEmpty()) {
+                throw new IllegalArgumentException("no predict action found");
+            }
+            String preProcessFunction = predictAction.get().getPreProcessFunction();
+            if (preProcessFunction != null && !MLPreProcessFunction.contains(preProcessFunction)) {
+                // user defined preprocess script, this case, the chunk size is always equals to text docs length.
+                return Tuple.tuple(textDocsLength, 1);
```

Comment:
Is this understanding correct ?
1. `textDocsLength` means how many chunks
2. `1` means step or chunk size ?

If correct, why hard code the chunk size as 1 ?  This issue https://github.com/opensearch-project/ml-commons/issues/2417 is an example, it's not the only case. For example some user may process two documents in one chunk.

Reply:
https://github.com/opensearch-project/ml-commons/blob/2.x/docs/remote_inference_blueprints/bedrock_connector_titan_multimodal_embedding_blueprint.md

Reply:
textDocsLength is chunks, 1 means step. When user process two documents in one chunk, user has to specify the `input_docs_processed_step_size`, in this case, this won't enter this branch. Multi-modal is a case that user needs to handle two documents in one chunk, I've tested this case and it works well.

Reply:
Thanks for testing multi-modal case. 
It's possible that step size > 1 and no `input_docs_processed_step_size`. For example, before async http client, user can confiture pre_process function to read three documents and construct `"inputs": [doc1, doc2, doc3]`, when response comes back, will move to next 3 docs

Reply:
This is a rare case, and we have workaround which is to add `input_docs_processed_step_size` configuration.

Reply:
Thanks, based on our discussion, this sounds good